runtime/common/src/lib.rs

```diff
@@ -58,3 +64,62 @@ parameter_types! {
 		.saturating_sub(Perbill::from_percent(10)) * MaximumBlockWeight::get();
 	pub const MaximumBlockLength: u32 = 5 * 1024 * 1024;
 }
+
+
+/// Apply a given filter to transactions.
+pub struct TransactionCallFilter<T: Filter<Call>, Call>(PhantomData<(T, Call)>);
+
+impl<F: Filter<Call>, Call> Default for TransactionCallFilter<F, Call> {
+	fn default() -> Self { Self::new() }
+}
+impl<F: Filter<Call>, Call> Encode for TransactionCallFilter<F, Call> {
+	fn using_encoded<R, FO: FnOnce(&[u8]) -> R>(&self, f: FO) -> R { f(&b""[..]) }
+}
+impl<F: Filter<Call>, Call> Decode for TransactionCallFilter<F, Call> {
+	fn decode<I: codec::Input>(_: &mut I) -> Result<Self, codec::Error> { Ok(Self::new()) }
+}
+impl<F: Filter<Call>, Call> Clone for TransactionCallFilter<F, Call> {
+	fn clone(&self) -> Self { Self::new() }
+}
+impl<F: Filter<Call>, Call> Eq for TransactionCallFilter<F, Call> {}
+impl<F: Filter<Call>, Call> PartialEq for TransactionCallFilter<F, Call> {
+	fn eq(&self, _: &Self) -> bool { true }
```

Comment:
What is the reasoning behind such an implementation?

Reply:
Using PhantomData